### PR TITLE
Add libRooFitMore: Move roofit file depending on GSL in a new library 

### PR DESCRIPTION
**Pull request by lmoneta**
Move the fRoofit classes depending on GSL in a new library libRooFitMore

The classes are: 

- AdaptiveGauss and GausKronrod 1D integrator from Roofitcore
- Legendre, non-central chi2 and SphArmonic pdf

### CODE REVIEW

File: test/CMakeLists.txt [hunk 1]
```diff
@@ -216,7 +216,11 @@ ROOT_ADD_TEST(test-stressmathcore-interpreted COMMAND ${ROOT_root_CMD} -b -q -l
 
 #--stressRooFit----------------------------------------------------------------------------------
 if(ROOT_roofit_FOUND)
-  ROOT_EXECUTABLE(stressRooFit stressRooFit.cxx LIBRARIES RooFit)
+if(ROOT_mathmore_FOUND)
+  ROOT_EXECUTABLE(stressRooFit stressRooFit.cxx LIBRARIES RooFit RooFitMore)
```
Comment (Axel-Naumann) [inline test/CMakeLists.txt]: This seems to be the same code in `if` and `else`?

File: roofit/roofitmore/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    src/RooFitMoreLib.cxx
+    src/RooAdaptiveGaussKronrodIntegrator1D.cxx
+    src/RooGaussKronrodIntegrator1D.cxx
+    src/RooLegendre.cxx
+    src/RooSpHarmonic.cxx
+    src/RooNonCentralChiSquare.cxx
+  DICTIONARY_OPTIONS
+    "-writeEmptyRootPCM"
+  LINKDEF
+    LinkDef1.h
```
Comment (Axel-Naumann) [inline roofit/roofitmore/CMakeLists.txt]: Can you rename that file to the more common name `LinkDef.h`?

File: roofit/roofitmore/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,52 @@
+# Copyright (C) 1995-2019, Rene Brun and Fons Rademakers.
+# All rights reserved.
+#
+# For the licensing terms see $ROOTSYS/LICENSE.
+# For the list of contributors see $ROOTSYS/README/CREDITS.
+
+############################################################################
+# CMakeLists.txt file for building ROOT roofitcore package
```
Comment (Axel-Naumann) [inline roofit/roofitmore/CMakeLists.txt]: `roofitcore` -> `roofitmore`

File: roofit/roofitmore/CMakeLists.txt [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+# @author Pere Mato, CERN
```
Comment (Axel-Naumann) [inline roofit/roofitmore/CMakeLists.txt]: `@author` should be you.

File: roofit/CMakeLists.txt [hunk 5]
```diff
@@ -6,6 +6,7 @@
 
 add_subdirectory(roofitcore)
 add_subdirectory(roofit)
+add_subdirectory(roofitmore)
```
Comment (Axel-Naumann) [inline roofit/CMakeLists.txt]: Should this not be protected by `if(mathmore)`?

File: roofit/roofitmore/CMakeLists.txt [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+############################################################################
+
+if(mathmore OR ROOT_mathmore_FOUND)
+  set(ROOT_MATHMORE_LIBRARY MathMore)
```
Comment (Axel-Naumann) [inline roofit/roofitmore/CMakeLists.txt]: How are the GSL symbols resolved if `mathmore` is off? Shouldn't all of `roofitmore` only be built if `mathmore` is on?

File: roofit/roofitmore/inc/LinkDef1.h [hunk 7]
```diff
@@ -0,0 +1,13 @@
+#ifdef __CINT__ 
+#pragma link off all globals;
+#pragma link off all classes;
+#pragma link off all functions;
+
+#pragma link C++ class RooAdaptiveGaussKronrodIntegrator1D+;
```
Comment (Axel-Naumann) [inline roofit/roofitmore/inc/LinkDef1.h]: This line should be removed from `roofit/roofitcore/inc/LinkDef1.h` else we have two entries for that class.

**Comment by lmoneta** [comment]
@phsft-bot build!

**Comment by lmoneta** [comment]
@phsft-bot build!

**Pull request merged**